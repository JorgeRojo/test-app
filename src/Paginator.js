import React from 'react';
import range from 'lodash.range';
import get from 'lodash.get';

const Paginator = ({ responseData, ...props }) => {
  const currentPage = get(responseData, 'page', 1);
  const totalPages = get(responseData, 'total_pages', 1);

  const handlePageChange = page => e => {
    e.preventDefault();
    props.handlePageChange(page);
  };

  return (
    <nav>
      <hr />
      <ul style={styles.list}>
        {range(1, totalPages + 1).map(page => (
          <li key={page} style={styles.listItem}>
            <a onClick={handlePageChange(page)} href={`#page_${page}`}>
              {currentPage === page ? <b>{page}</b> : <span>{page}</span>}
            </a>
          </li>
        ))}
      </ul>
      <hr />
    </nav>
  );
};

const styles = {
  list: { listStyle: 'none' },
  listItem: { display: 'inline-block', padding: '10px' },
};

export default Paginator;
