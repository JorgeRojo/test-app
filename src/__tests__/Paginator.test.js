import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import Paginator from '../Paginator';

describe('Paginator.js', () => {
  let container = null;

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('renders without crashing', () => {
    render(<Paginator />, container);
    unmountComponentAtNode(container);
  });

  it('renders data changes', () => {
    act(() => {
      const responseData = { page: 1, total_pages: 10 };
      render(<Paginator {...{ responseData }} />, container);
    });
    expect(container).toMatchSnapshot();

    act(() => {
      const responseData = { page: 2, total_pages: 5 };
      render(<Paginator {...{ responseData }} />, container);
    });
    expect(container).toMatchSnapshot();
  });

  it('page changes when clicked', () => {
    const responseData = { page: 1, total_pages: 5 };
    const handlePageChange = jest.fn();

    act(() => {
      render(<Paginator {...{ responseData, handlePageChange }} />, container);
    });

    const button = document.querySelector(`a[href="#page_3"]`);
    expect(button.textContent).toBe('3');

    act(() => {
      button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });

    expect(handlePageChange).toHaveBeenCalledTimes(1);
    expect(handlePageChange).toHaveBeenCalledWith(3);
  });
});
