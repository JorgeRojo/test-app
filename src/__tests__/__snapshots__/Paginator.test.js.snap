// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator.js renders data changes 1`] = `
<div>
  <nav>
    <hr />
    <ul
      style="list-style: none;"
    >
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_1"
        >
          <b>
            1
          </b>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_2"
        >
          <span>
            2
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_3"
        >
          <span>
            3
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_4"
        >
          <span>
            4
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_5"
        >
          <span>
            5
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_6"
        >
          <span>
            6
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_7"
        >
          <span>
            7
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_8"
        >
          <span>
            8
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_9"
        >
          <span>
            9
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_10"
        >
          <span>
            10
          </span>
        </a>
      </li>
    </ul>
    <hr />
  </nav>
</div>
`;

exports[`Paginator.js renders data changes 2`] = `
<div>
  <nav>
    <hr />
    <ul
      style="list-style: none;"
    >
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_1"
        >
          <span>
            1
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_2"
        >
          <b>
            2
          </b>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_3"
        >
          <span>
            3
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_4"
        >
          <span>
            4
          </span>
        </a>
      </li>
      <li
        style="display: inline-block; padding: 10px;"
      >
        <a
          href="#page_5"
        >
          <span>
            5
          </span>
        </a>
      </li>
    </ul>
    <hr />
  </nav>
</div>
`;
